pennylane/templates/embeddings/amplitude.py

```diff
@@ -204,6 +204,13 @@ def _preprocess(features, wires, pad_with, normalize):
             # pad
             if pad_with is not None and n_features < 2 ** len(wires):
                 padding = [pad_with] * (2 ** len(wires) - n_features)
+                if (
+                    hasattr(feature_set, "device") and feature_set.device.type == "cuda"
+                ):  # pragma: no cover
+                    ## Torch tensor, send to same GPU
+                    dat_type = type(feature_set)
```

Comment:
How come the type is queried here and used to convert `padding`? Would using `to(feature_set.device)` not suffice?

Reply:
The idea here was since `padding` begins life as a Python list, we need to explicitly create the Torch Tensor type to match it. We could instead create `padding` as a separate Tensor directly inside the if-statement, and use `.to` here, if you think that would be more sensible?

Reply:
Oh, that makes sense! No, this is great. :+1: